# 01 — Decision-Rule Gap Audit: Claim Traceability

**EPISTEMIC STATUS: EXPLANATORY**

This notebook renders and walks through the decision-rule gap audit — the core evidential contribution of Paper 1. It loads structured data extracted from the manuscript's Table 1 and provides a transparent, inspectable presentation of the audit logic.

**Claim IDs (primary coverage in this notebook):** P1-C01, P1-C02, P1-C03, P1-C08, P1-C09, P1-C10, P1-C11, P1-C12, P1-C22 (limitations / audit scope; see matrix for PARTIAL vs TRACEABLE).

**STM targets:** STM-T1 (Table 1), STM-MA4 (Coding Frame reference), STM-MA5 (Evidence Table reference)

**Outputs:** `outputs/tables/table1_rendered.csv`


## Context: What Is the Decision-Rule Gap Audit?

Paper 1 conducts a targeted document audit of widely used clinical AI governance instruments. The audit assesses whether each instrument:

1. Specifies an explicit rule for converting multi-domain governance evidence into a binary deployment decision (go/no-go)
2. Defines aggregation logic (e.g., weighted scoring or minimum floors)
3. Specifies any override mechanism and constraints
4. Explicitly assigns final deployment authority

The audit was limited to explicit statements in the instrument text; items were coded as "present" only when decision logic was described in operational terms (e.g., thresholds, pass/fail criteria, or a defined aggregation rule), rather than as high-level recommendations or general lifecycle principles.

The full coding frame is documented in **Multimedia Appendix 4** (`supplementary/Multimedia_Appendix_4_AuditCodingFrame.docx`). The full evidence table with verbatim excerpts and coding decisions is in **Multimedia Appendix 5** (`supplementary/Multimedia_Appendix_5_EvidenceTable.docx`).


In [1]:
import sys
from pathlib import Path

REPO_ROOT = Path.cwd().parent if Path.cwd().name == "notebooks" else Path.cwd()
sys.path.insert(0, str(REPO_ROOT / "src"))

from p1_harness.table1_gap_audit import load_table1_data, validate_table1_schema

table1_data = load_table1_data(REPO_ROOT)
validate_table1_schema(table1_data)

print(
    f"Schema validation PASSED: {len(table1_data['rows'])} frameworks × {len(table1_data['columns'])} dimensions"
)
print(f"Source: {table1_data['metadata']['source']}")


Schema validation PASSED: 7 frameworks × 5 dimensions
Source: Table 1, Paper1_Manuscript.docx


In [2]:
import pandas as pd
from p1_harness.table1_gap_audit import render_table1_dataframe, write_table1_csv

# Render Table 1 as formatted DataFrame
df = render_table1_dataframe(table1_data)
output_path = write_table1_csv(REPO_ROOT, df)
print(f"Written: {output_path.relative_to(REPO_ROOT)}")

# Display
df.style.set_caption(
    "Table 1. Decision-rule elements explicitly stated in selected clinical AI governance instruments"
)


Written: outputs\tables\table1_rendered.csv


,Framework,Explicit deployment decision rule,Aggregation logic specified,Override pathway specified,Final deployment authority assigned
0,FUTURE-AI [26],Not specified,Not specified,Not specified,Partial (process ownership)
1,TEHAI [28],Not specified,Not specified,Not specified,Not specified
2,NICE ESF [5],Not specified,Not specified,Not specified,Partial
3,FDA AI-DSF [27],Not applicable (regulatory submission),Not specified,Not specified,Partial
4,EU AI Act [3],Not specified as institutional decision rule,Not specified,Not specified,Yes (provider/deployer obligations)
5,ISO/IEC 42001 [4],Not specified,Not specified,Not specified,Yes (management system owner)
6,This Viewpoint,Specified,Specified (minimum floors),Specified (constrained),Specified


## Reading the Table

**Rows 1–6 (existing instruments):** Across all six reviewed instruments, the dominant pattern is:

- **Explicit deployment decision rule:** Not specified in any reviewed instrument as an institutional decision rule
- **Aggregation logic:** Not specified in any reviewed instrument
- **Override pathway:** Not specified in any reviewed instrument  
- **Final deployment authority:** Partial or framework-specific in most; only the EU AI Act and ISO 42001 assign explicit authority (to provider/deployer and management system owner, respectively)

**Row 7 (This Viewpoint):** The proposed gate architecture specifies all four elements — deployment decision rule, aggregation logic (minimum floors), override pathway (constrained), and deployment authority.

### Key Finding

The audit supports a narrow but practically important inference: within the reviewed instruments, the dominant pattern is specification of *what* to evaluate and what lifecycle controls to maintain, with comparatively limited specification of *how* heterogeneous evidence should be operationally combined into a binding deployment decision.


## Interpretive Caveat

> "All characterisations are the author's interpretive assessment."
> — Paper 1 Manuscript

The audit was conducted by a single author using a conservative coding frame (items coded as "present" only when operational decision logic was explicitly stated). This conservatism may overestimate the gap if decision rules exist in sections not audited. The full evidence table (Multimedia Appendix 5) provides transparency for readers to assess the coding decisions independently.

## Cross-References

- **Multimedia Appendix 4:** Coding frame and search strategy (`supplementary/Multimedia_Appendix_4_AuditCodingFrame.docx`)
- **Multimedia Appendix 5:** Full evidence table with verbatim excerpts (`supplementary/Multimedia_Appendix_5_EvidenceTable.docx`)


In [3]:
import hashlib

# Output hash for manifest validation
h = hashlib.sha256()
with open(output_path, "rb") as f:
    h.update(f.read())
print(f"SHA-256 of {output_path.name}: {h.hexdigest()}")


SHA-256 of table1_rendered.csv: 578c4583a341c5aa20bde05bdcfdb1bf50dfe940c3bdf1e248263d4ec722079f
